src/utils/queue_builder.rs

```diff
@@ -17,6 +19,8 @@ pub fn queue_builder(
             report_error(&anyhow::anyhow!(e).context("failed to remove temporary directories"));
         }
 
+        builder = builder.peroid_init(context)?;
```

Comment:
we should decide what to do when the re-init fails with any error. 

( whereever the reinit will end up, here or inside `RustwideBuilder`). 

Either we just long an error and continue building crates, though I lean towards locking the queue as we do when the toolchain update fails. 

In any case, just returning with `?` is not good in this place.

Reply:
i split the code to `build_workspace` which build workspace, and then i rename the `peroid_init` to `reinitialize_workspace_if_interval_passed`, as the name shown, the new function will reinitialize workspace if the interval passed.